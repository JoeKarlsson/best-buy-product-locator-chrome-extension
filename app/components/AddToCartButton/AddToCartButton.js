import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { COLORS, STYLES } from '../../constants/styles';

const Container = styled.div`
  ${(props) => {
    if (!props.isPopup) {
      return css`
        position: absolute;
        width: 200px;
        right: 60px;
        top: 10px;

        @media (max-width: 1110px) {
          margin-top: 10px;
          position: initial;
          float: left;
        }

        @media (max-width: 710px) {
          margin-top: 15px;
          width: 100%;
        }
      `;
    }
  }};
`;

const Button = styled.button`
  color: ${COLORS.black};
  background: ${COLORS.yellow};
  font-family: ${STYLES.fontFamily};
  font-size: ${STYLES.fontSize};
  width: 100%;
  border: 0;
  height: 45px;
  line-height: normal;
  cursor: pointer;
  border-radius: 4px;
  font-weight: 700;
  text-align: center;
  vertical-align: middle;
  display: inline-block;
  transition: all 0.3s ease-out 0s;

  &:hover {
    background: ${COLORS.lightYellow};
  }
`;

const AddToCartButton = ({ addToCartUrl, isPopup }) => (
  <Container isPopup={isPopup}>
    <a target="_blank" rel="noopener noreferrer" href={addToCartUrl}>
      <Button>Buy on BestBuy.com</Button>
    </a>
  </Container>
);

AddToCartButton.propTypes = {
  addToCartUrl: PropTypes.string.isRequired,
  isPopup: PropTypes.bool,
};

AddToCartButton.defaultProps = {
  isPopup: false,
};

export default AddToCartButton;
