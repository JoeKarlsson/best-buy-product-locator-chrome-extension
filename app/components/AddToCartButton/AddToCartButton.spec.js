import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import AddToCartButton from './AddToCartButton';

configure({ adapter: new Adapter() });

describe('AddToCartButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AddToCartButton addToCartUrl="www.testurl.com" />);
  });

  describe('rendering', () => {
    describe('initial state', () => {
      it('is rendered', () => {
        const component = renderer.create(<AddToCartButton addToCartUrl="www.testurl.com" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('is rendered correctly', () => {
        expect(wrapper).toHaveLength(1);
      });

      it('should have correct inital instance', () => {
        const initialInstance = wrapper.instance();
        const expectedInstance = null;
        expect(initialInstance).toBe(expectedInstance);
      });
    });
  });
});
