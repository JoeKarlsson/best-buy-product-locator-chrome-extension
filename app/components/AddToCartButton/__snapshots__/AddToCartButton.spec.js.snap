// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton rendering initial state is rendered 1`] = `
<div
  className="sc-bdVaJa jOColq"
>
  <a
    href="www.testurl.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <button
      className="sc-bwzfXH gQLsoL"
    >
      Buy on BestBuy.com
    </button>
  </a>
</div>
`;
